xbmc/windowing/gbm/drm/DRMUtils.cpp

```diff
@@ -150,10 +150,11 @@ bool CDRMUtils::FindPreferredMode()
 
     if(current_mode->type & DRM_MODE_TYPE_PREFERRED)
     {
-      m_mode = current_mode;
       CLog::Log(LOGDEBUG, "CDRMUtils::{} - found preferred mode: {}x{}{} @ {} Hz", __FUNCTION__,
-                m_mode->hdisplay, m_mode->vdisplay,
-                m_mode->flags & DRM_MODE_FLAG_INTERLACE ? "i" : "", m_mode->vrefresh);
+                current_mode->hdisplay, current_mode->vdisplay,
+                current_mode->flags & DRM_MODE_FLAG_INTERLACE ? "i" : "", current_mode->vrefresh);
+      if (!m_mode)
```

Comment:
Can you pull this out of the loop completely and move it to the beginning of the method and just return true if m_mode is set?

Reply:
done